src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
 
+const DEFAULT_HEALTH = 100;
+const DEFAULT_DEMAGE = 50;
+
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = DEFAULT_HEALTH) {
+    this.name = name;
+    this.health = DEFAULT_HEALTH;
+    Animal.alive.push(this);
+  }
+
+  demage(amout = DEFAULT_DEMAGE) {
+    this.health -= amout;
+
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((item) => item !== this);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim instanceof Carnivore || victim.hidden) {
+      return;
+    }
+
+    victim.demage(DEFAULT_DEMAGE);
```

Comment:
do the action only when 2 conditions are met.
u can shorten ur code
```suggestion
    if (victim instanceof Herbivore && !victim.hidden) {
       victim.demage(DEFAULT_DEMAGE);
    }
```